src/components/settings/settings-custom-labels/settings-custom-labels.js

```diff
@@ -23,197 +26,183 @@ import SettingsForm from '../settings-form';
 import NavigationModal from '../../navigation-modal';
 import { formatCustomLabelsValues } from '../../utilities';
 
-class SettingsCustomLabels extends Component {
-  static propTypes = {
-    confirmUpdate: PropTypes.func.isRequired,
-    credentialId: PropTypes.string.isRequired,
-    customLabels: PropTypes.shape({
-      errors: PropTypes.array,
-      isUpdated: PropTypes.bool,
-      items: PropTypes.shape({
-        data: PropTypes.arrayOf(PropTypes.shape({
-          displayLabel: PropTypes.string,
-          displayOnFullTextFinder: PropTypes.bool,
-          displayOnPublicationFinder: PropTypes.bool,
-          id: PropTypes.number,
-        })),
-      }),
-    }).isRequired,
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-    updateCustomLabels: PropTypes.func.isRequired,
-  };
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      modalIsOpen: false,
-      removingLabels: '',
-    };
-  }
+const SettingsCustomLabels = ({
+  confirmUpdate,
+  credentialId,
+  customLabels,
+  updateCustomLabels,
+}) => {
+  const intl = useIntl();
 
-  componentDidUpdate() {
-    const {
-      confirmUpdate,
-      customLabels: { isUpdated },
-    } = this.props;
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [removingLabels, setRemovingLabels] = useState('');
 
-    if (isUpdated) {
+  useEffect(() => {
+    if (customLabels.isUpdated) {
       confirmUpdate();
     }
-  }
-
-  prepareInitialValues = () => {
-    const { customLabels: { items: { data } } } = this.props;
-
-    return data ? data.reduce((acc, { attributes }) => {
-      const {
-        id,
-        displayLabel,
-        displayOnFullTextFinder,
-        displayOnPublicationFinder,
-      } = attributes;
-
-      return {
-        ...acc,
-        [`customLabel${id}`]: {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [customLabels]);
```

Comment:
You should pass `customLabels.isUpdated` as a dependency. Passing just `customLabels` will cause the callback to be called too many times when a reference to `customLabels` has changed, but `isUpdated` value is still the same